docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,144 @@
+initial forcing data for the model - things like air temperature, elevation and
+photosynthetically active radiation - but also  includes the storage of internal
+variables calculated by the various models running within the simulation. The data
+handling for simulations is managed by the {mod}`~virtual_rainforest.core.data` module
+and the {class}`~virtual_rainforest.core.data.Data` class, which provides the data
+loading and storage functions for the Virtual Rainforest. The data system is extendable
+to provide support for different file formats and axis validation (see the [module API
+docs](../../api/core/data.md)) but that is beyond the scope of this document.
+
+A Virtual Rainforest simulation will have one instance of the
+{class}`~virtual_rainforest.core.data.Data` class to provide access to the different
+forcing and internal variables used in the simulation. As they are loaded, all variables
+are validated and then  added to an {class}`xarray.Dataset` object, which provides a
+consistent indexing and data manipulation for the underlying arrays of data.
+
+In many cases, a user will simply provide a configuration file to set up the data that
+will be validated and loaded when a simulation runs, but the main functionality for
+working with data using Python are shown below.
+
+## Validation
+
+One of the main functions of the {mod}`~virtual_rainforest.core.data` module is to
+automatically validate data before it is added to the `Data` instance. Validation is
+applied along a set of **core axes** used in the simulation. Each core axis has a set of
+validators: each validator in the set detects a possible data configuration and then
+runs code to validate data in that configuration. The validation process is primarily
+intended to check that provided data is congruent with the configuration of a particular
+simulation and to map the data onto a core axis.
+
+The validators use the dimension names of input data to detect if that data should be
+validated on a particular axis. For example, the `x` and `y` dimension names are used to
+trigger validation on the `spatial` core axis.
+
+For more details on the different core axes and the alternative mappings applied by
+validators see the [core axis](axes.md) documentation.
+
+## Creating a `Data` instance
+
+A  {class}`~virtual_rainforest.core.data.Data` instance is created using information
+that provides information on the core configuration of the simulation. At present, this
+is just the spatial grid being used.
+
+```{code-cell} ipython3
+from virtual_rainforest.core.grid import Grid
+from virtual_rainforest.core.data import Data
+from virtual_rainforest.core.axes import *
+from xarray import DataArray
+import numpy as np
+
+# Create a simple default grid and a Data instance
+grid = Grid()
+data = Data(grid=grid)
+
+data
+```
+
+## Adding data to a Data instance
+
+Data can be added to a {class}`~virtual_rainforest.core.data.Data` instance using one of
```

Comment:
Would be good to have some details about when these different methods should be used, also isn't completely clear to me why only the first method gets an extended description here. Is it that it is the method we expect people to mostly use?

Reply:
No - the others are just wrappers around this but we'd typically expect people to use `load_from_file` and `load_data_config`. I've added examples.